### PR TITLE
fix to swallow attribute on brace expression

### PR DESCRIPTION
**Pull request by rchaser53**
fix: #3845

### CODE REVIEW

File: src/closures.rs [hunk 1]
```diff
@@ -103,8 +103,21 @@ fn get_inner_expr<'a>(
 
 // Figure out if a block is necessary.
 fn needs_block(block: &ast::Block, prefix: &str, context: &RewriteContext<'_>) -> bool {
+    let has_attribute = if let Some(first_stmt) = block.stmts.first() {
+        match first_stmt.kind {
```
Comment (topecongiro) [inline src/closures.rs]: I almost forgot, but looks like we have `get_attrs_from_stmt` to extract attributes from `ast::Stmt` :laughing:

Comment (topecongiro) [reply src/closures.rs]: Nit: using `map_or` looks cleaner IMHO:

```rust
let has_attributes = block.stmts.first().map_or(false, |first_stmt| {
    !get_attrs_from_stmt(first_stmt).is_empty()
});
```

Comment (rchaser53) [reply src/closures.rs]: Actually I searched something good utily method, but I failed to find get_attrs_from_stmt. Thank you.

**Comment by topecongiro** [comment]
Thank you for the update 🎉

**Pull request merged**